datadog/resource_datadog_logs_metric.go

```diff
@@ -0,0 +1,324 @@
+package datadog
+
+import (
+	"fmt"
+
+	datadogV2 "github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceDatadogLogsMetric() *schema.Resource {
+	return &schema.Resource{
+		Description: "Resource for interacting with the logs_metric API",
+		Create:      resourceDatadogLogsMetricCreate,
+		Read:        resourceDatadogLogsMetricRead,
+		Update:      resourceDatadogLogsMetricUpdate,
+		Delete:      resourceDatadogLogsMetricDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceDatadogLogsMetricImport,
+		},
+		Schema: map[string]*schema.Schema{
+
+			"compute": {
+				Type:        schema.TypeList,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The compute rule to compute the log-based metric. This field can't be updated after creation.",
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"aggregation_type": {
+							Type:        schema.TypeString,
```

Comment:
Since this is an enum, can you add the `ValidateFunc: validateEnumValue(datadogV2.NewLogsMetricComputeAggregationTypeFromValue)` in here, instead of in the `getCompute` method below ?

Reply:
Good point! Thanks 👍